docs/tools/terraform/reference/cookbook/kafka-mongodb-recipe.rst

```diff
@@ -0,0 +1,86 @@
+Apache Kafka® with MongoDB Source Connector
+===========================================
+
+The `Aiven Terraform Provider <https://registry.terraform.io/providers/aiven/aiven/latest/docs>`_ is a great choice for declaratively provisioning an Aiven for Apache Kafka® cluster with Kafka Connect enabled and the MongoDB source connector configured.
+
+Let's check out the following diagram to understand the setup.
+
+.. mermaid::
+
+   flowchart LR
+      id1[(MongoDB Database)]
+      subgraph Kafka Connect
+      MongoDB-Source-Connector
+      end
+      subgraph Apache Kafka
+      Topic
+      end
+      id1 --->|polls changes| MongoDB-Source-Connector -->|publishes changes| Topic
+
+Describe the setup
+------------------
+
+Here is the Terraform recipe that will spin up an Aiven for Apache Kafka service with Kafka Connect enabled. This recipe will also create and configure a MongoDB source connector. 
+Before you begin, you will require a MongoDB database and the related database connection information. You'll also need to make sure that the database is reachable from the public internet.
+
+.. Tip::
+
+    Be sure to check out the :doc:`getting started guide <../../get-started>` to learn about the common files required to execute the following recipe. For example, you'll need to declare the variables for ``project_name``, ``api_token``, ``service_name_prefix``, and ``mongodb_connection_uri``.
+
+``services.tf`` file:
+
+.. code:: terraform
+
+  resource "aiven_kafka" "demo-kafka" {
+    project                 = var.project_name
+    cloud_name              = "google-northamerica-northeast1"
+    plan                    = "business-4"
+    service_name            = join("-", [var.service_name_prefix, "kafka"])
+    maintenance_window_dow  = "sunday"
+    maintenance_window_time = "10:00:00"
+    kafka_user_config {
+      kafka_connect = true
+      kafka_rest    = true
+      kafka_version = "3.1"
+      kafka {
+        group_max_session_timeout_ms = 70000
```

Comment:
why do we have this custom settings? Are they strictly needed?

Reply:
They are not; removed. Good call.